### PR TITLE
Extract noData functionality

### PR DESCRIPTION
**Pull request by jamesgarfield**
Ref #787 

After fixing the chart clearing bug, I thought I should capitalize on my current familiarity of the 'noData' logic to extract it into it's own utils function.

Each chart still has it's own tests for 'noData', but I've also added one in the utils unit tests for safety.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -592,3 +592,39 @@ Calculate the available width for a chart.
...
     return nv.utils.sanitizeWidth(width, container) - margin.left - margin.right;
 }
+
+/*
+Clear any rendered chart components and display a chart's 'noData' message
+*/
+nv.utils.noData = function(chart, container) {
+    var opt = chart.options(),
+        margin = opt.margin(),
+        noData = opt.noData(),
```
Comment (liquidpele) [inline src/utils.js]: Lets do noData = opt.noData() || "No Data Available", and set the default of noData to null in the rest of the models... that way the default string is only defined in one place.

File: src/utils.js [hunk 2]
```diff
@@ -592,3 +592,39 @@ Calculate the available width for a chart.
...
+        .attr('y', y)
+        .text(nv.utils.identity);
+}
+
+/*
+Identity function
+*/
+nv.utils.identity = function(i) { 
+    return i; 
+}
```
Comment (liquidpele) [inline src/utils.js]: This function seems silly to be broken out by itself when it's only called in once place, just make it an anonymous function above like the original code was doing.

Comment (jamesgarfield) [inline src/utils.js]: I disagree about the utility of this, but I'm happy to leave it for another time.

**Comment by liquidpele** [comment]
Looks good!

**Pull request merged**